app/controllers/payments_controller.rb

```diff
@@ -0,0 +1,58 @@
+class PaymentsController < ApplicationController
+
+  before_action :authenticate_user!
+  load_and_authorize_resource
+  before_action :find_invoice
+
+  def index
+    @payments = @invoice.payments
+  end
+
+  def new
+    @payment = Payment.new
+  end
+
+  def edit
+    @payment = Payment.find(params[:id])
+  end
+
+  def create
+    @payment = @invoice.payments.build(payment_params)
+    @payment.date_of_payment = format_date_locale(params[:payment][:date_of_payment])
+    if @payment.save
+      begin
+        ClientMailer.send_email_payment(@invoice, current_user).deliver if params[:email].present?
+        flash[:success] = 'Payment saved successfully.'
+        redirect_to invoice_payments_path
+      rescue
+        flash[:error] = "Problem while sending email."
+        render :action => 'new'
+      end
+    else
+      flash[:error] = "Problem while saving payment details. #{@payment.errors.full_messages.join(',')}"
+      render :action => 'new'
+    end
+  end
+
+  def find_invoice
+    @invoice = Invoice.find(params[:invoice_id])
+  end
+
+  def update
+    params_date_of_payment = ActionController::Parameters.new(date_of_payment: format_date_locale(params[:payment][:date_of_payment])).permit(:date_of_payment)
+    if @payment.update(payment_params)
+      @payment.update(params_date_of_payment)
```

Comment:
This update can be avoided. For that you have to replace `params[:payment][:date_of_payment]` with formatted date. Rest will be taken care by rails